account_invoice_inter_company/models/res_company.py

```diff
@@ -16,7 +16,7 @@ class ResCompany(models.Model):
 
     def _compute_share_product(self):
         self.ensure_one()
-        product_rule = self.env.ref('product.product_comp_rule')
+        product_rule = self.sudo().env.ref('product.product_comp_rule')
```

Comment:
Don't do this. Instead, put `compute_sudo=True` in the field definition. FYI, in v13 all compute are by default this way.